src/js/tracks/text-track-settings.js

```diff
@@ -290,21 +306,28 @@ class TextTrackSettings extends Component {
    *         The DOM element that gets created.
    * @private
    */
-  createElSelect_(key) {
+  createElSelect_(key, legendId = '') {
     const config = selectConfigs[key];
     const id = config.id.replace('%s', this.id_);
 
     return [
       createEl('label', {
+        id,
         className: 'vjs-label',
         textContent: this.localize(config.label)
       }, {
-        for: id
       }),
-      createEl('select', {id}, undefined, config.options.map(o => {
+      createEl('select', {}, {
+        'aria-labelledby': `${legendId} ${id}`
+      }, config.options.map(o => {
+        const optionId = id + '-' + o[1];
+
         return createEl('option', {
+          id: optionId,
           textContent: this.localize(o[1]),
           value: o[0]
+        }, {
+          'aria-labelledby': `${legendId} ${id} ${optionId}`
```

Comment:
I'm a little concerned about how well this will work - there's a circular reference where the label for this element includes the `id` of two other elements, and then the **id of this element**. The behavior in this situation *is mentioned* in the W3C [Accessible Name Computation - Text Alternative Computation](https://www.w3.org/TR/accname-aam-1.1/#mapping_additional_nd_te):

> if computing a name, and the current node has a non-empty aria-labelledby attribute, _and the current node is not already part of an aria-labelledby traversal_, process its IDREFs in the order they occur: 

but it may not work correctly in all browsers. I guess we'll just have to fix it if that happens?